bootstrap-salt.sh

```diff
@@ -1813,27 +1819,26 @@ __function_defined() {
 #                 a boot process, such as on AWS AMIs. This func will wait until the boot
 #                 process is finished so the script doesn't exit on a locked proc.
 #----------------------------------------------------------------------------------------------------------------------
+APT_ERR=$(mktemp /tmp/apt_error.XXXX)
 __wait_for_apt(){
     # Timeout set at 15 minutes
     WAIT_TIMEOUT=900
 
     # Run our passed in apt command
-    "${@}"
+    "${@}" 2>$APT_ERR
     APT_RETURN=$?
 
-    # If our exit code from apt is 100, then we're waiting on a lock
-    while [ $APT_RETURN -eq 100 ]; do
+    # Make sure we're not waiting on a lock
+    while [ $APT_RETURN -ne 0 ] && grep -q -v '^E: Could not get lock' "$APT_ERR"; do
       echoinfo "Aware of the lock. Patiently waiting $WAIT_TIMEOUT more seconds..."
       sleep 1
       WAIT_TIMEOUT=$((WAIT_TIMEOUT - 1))
 
-      # If timeout reaches 0, abort.
       if [ "$WAIT_TIMEOUT" -eq 0 ]; then
           echoerror "Apt, apt-get, aptitude, or dpkg process is taking too long."
           echoerror "Bootstrap script cannot proceed. Aborting."
           return 1
       else
-          # Try running apt again until our return code != 100
 	  "${@}"
```

Comment:
What missing here is `stderr` redirect to temporary log file, because for now we only check the message for the very first `apt` invocation.

Reply:
True. Fixed.